lib/core/managementClient.ts

```diff
@@ -532,6 +556,209 @@ export class ManagementClient extends LinkEntity {
     }
   }
 
+
+  /**
+   * Get all the rules on the Subscription.
+   * @returns Promise<RuleDescription> A list of rules.
+   */
+  async getRules(): Promise<RuleDescription[]> {
+    try {
+      const request: AmqpMessage = {
+        body: {
+          top: types.wrap_int(max32BitNumber),
+          skip: types.wrap_int(0)
+        },
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.enumerateRules
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+
+      log.mgmt("[%s] Get rules request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      const response = await this._mgmtReqResLink!.sendRequest(request);
+      if (response.application_properties!.statusCode === 204 || !response.body || !Array.isArray(response.body.rules)) {
+        return [];
+      }
+
+      // Reference: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#response-11
+      const result: { "rule-description": Typed }[] = response.body.rules || [];
+      const rules: RuleDescription[] = [];
+      result.forEach((x) => {
+        const ruleDescriptor = x['rule-description'];
+        if (!Array.isArray(ruleDescriptor.value) || ruleDescriptor.value.length < 3) {
+          return;
+        }
+        const filtersRawData: Typed = ruleDescriptor.value[0];
+        const actionsRawData: Typed = ruleDescriptor.value[1];
+        const rule: RuleDescription = {
+          name: ruleDescriptor.value[2].value
+        };
+
+        switch (filtersRawData.descriptor.value) {
+          case 83483426823:
```

Comment:
Could we turn these into constants that are easier to understand?

Reply:
Yes, but these should live in the amqp-common library. So I have a PR out to that repo. See https://github.com/Azure/amqp-common-js/pull/13

Reply:
Great! Before that PR is accepted, could you use the same code in here? Once the PR is merged and accepted you won't have to touch the code, just consume the new version of the package.

Reply:
We could create new constants for now in this sdk. But we still have to replace them with references to the ones from the amqp library once a new version is released. There is no way we dont have to touch the code again if we want to use the constants from the amqp library

Reply:
Fixed